internal/status/collector/handle.go

```diff
@@ -30,32 +26,21 @@ const (
 func HandleReconcileStatus(ctx context.Context, log logr.Logger, params manifests.Params, otelcol v1beta1.OpenTelemetryCollector, err error) (ctrl.Result, error) {
 	log.V(2).Info("updating collector status")
 	if err != nil {
-		params.Recorder.Event(&otelcol, eventTypeWarning, reasonError, err.Error())
+		params.Recorder.Event(&otelcol, corev1.EventTypeWarning, reasonError, err.Error())
 		return ctrl.Result{}, err
 	}
-	changed := otelcol.DeepCopy()
 
-	up := &collectorupgrade.VersionUpgrade{
-		Log:      params.Log,
-		Version:  version.Get(),
-		Client:   params.Client,
-		Recorder: params.Recorder,
-	}
-	upgraded, upgradeErr := up.ManagedInstance(ctx, *changed)
```

Comment:
Not sure why the upgrade was called in the `HandleReconcileStatus()` function?

My best guess is to update the `Status.Version` field? But that's already set in `updateCollectorStatus()`, which is called in `HandleReconcileStatus()`.
Also, the upgrade ran here, but it did not update the (modified) CR in the cluster.

Reply:
I think it was to set the status field, yes. @jaronoff97 you refactored this back in 2023, do you recall if this is correct?

Reply:
yes, this was to set the status. I believe this was also to keep the functionality the same during my refactor.